akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -80,13 +80,13 @@ abstract class PathMatcher[L](implicit val ev: Tuple[L]) extends (Path ⇒ PathM
     new PathMatcher[lift.Out]()(lift.OutIsTuple) {
       require(min >= 0, "`min` must be >= 0")
       require(max >= min, "`max` must be >= `min`")
-      def apply(path: Path) = rec(path, 1)
+      def apply(path: Path) = rec(path, 0)
       def rec(path: Path, count: Int): Matching[lift.Out] = {
         def done = if (count >= min) Matched(path, lift()) else Unmatched
-        if (count <= max) {
+        if (count < max) {
           self(path) match {
             case Matched(remaining, extractions) ⇒
-              def done1 = if (count >= min) Matched(remaining, lift(extractions)) else Unmatched
+              def done1 = if (count + 1 >= min) Matched(remaining, lift(extractions)) else Unmatched
```

Comment:
here checking if **count+1 >= min instead of count >= min**, because we `self(path)` already matched a pattern and we need to increment count. (as mentioned earlier count denotes number of times the expected pattern appeared in the path till now. )